src/Microsoft.Health.SqlServer/Features/Schema/SchemaJobWorker.cs

```diff
@@ -71,10 +71,12 @@ public async Task ExecuteAsync(SchemaInformation schemaInformation, string insta
                         schemaInformation.Current = await schemaDataStore.UpsertInstanceSchemaInformationAsync(instanceName, schemaInformation, cancellationToken);
 
                         // If there was a change in the schema version
-                        if (previous != schemaInformation.Current)
+                        if (schemaInformation.Current != previous)
                         {
+                            var isFullSchemaSnapshot = previous == null;
```

Comment:
As discussed with @rbans96, I made changes in 10c011e so that we are no longer firing a `SchemaUpgradedNotification` when the base schema script is run.